### DISCUSSION TITLE
How to make a field not "Required"

### DISCUSSION THREAD

**Discussion by benjamin-kirkbride** [pydantic/pydantic#7104] [Pydantic V2]
To be extremely clear, I am not talking about making a field None/Nullable, I am asking how to make a field able to be not **required**

Take the following model:
```python
class Foo(BaseModel):
    not_required: int
    required_1: int
    required_2: int

foo = Foo(required_1 = 1, required_2 = 2)
```

What do I have to do to have that not result in an error?

The documentation helpfully includes three ways to mark a field as required, but zero ways to mark a field as not required.

https://docs.pydantic.dev/latest/usage/models/#required-fields

**Comment by Viicos** [comment]
I don't understand, if you don't give `not_required` a default value, what value should it take then? Are you looking for something similar to [`typing.NotRequired`](https://docs.python.org/3/library/typing.html#typing.NotRequired)? If so, you could use a `TypeAdapter` with a `TypedDict` (see [here](https://docs.pydantic.dev/latest/why/#typeddict)). I don't think this is supported yet with `BaseModel`
